app/scripts/metamask-controller.js

```diff
@@ -1699,7 +1696,6 @@ export default class MetamaskController extends EventEmitter {
       initState.SmartTransactionsController,
     );
 
-    ///: BEGIN:ONLY_INCLUDE_IF(petnames)
     const isExternalNameSourcesEnabled = () =>
```

Comment:
Sorry if am missing the obvious. How does the app is going to be reactive to this change if we are initialising `Providers` here?

If user toggle off the `useExternalNameSources`, shouldn't we need to tell controller to disable providers?

Reply:
Good question. I created [a separate ticket](https://github.com/orgs/MetaMask/projects/47/views/1?pane=issue&itemId=50429379) for this.

Reply:
This entire function reference is passed to the relevant providers as a callback meaning it's checked on every request, not just during initialisation.

Reply:
Oh that's right, missed that function going deep directly to providers. I instantly thought `isEnabled` as a boolean, so my mistake

Reply:
I missed this too. Resolving.